src/ResetPasswordHelper.php

```diff
@@ -110,6 +110,10 @@ public function validateTokenAndFetchUser(string $fullToken): object
     {
         $this->resetPasswordCleaner->handleGarbageCollection();
 
+        if (40 !== \strlen($fullToken)) {
```

Comment:
as the full token is supposed to be of known length, exit early if it is not 40 char long.